controllers/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -247,6 +255,19 @@ func (r *ReconcileUpgradeConfig) Reconcile(ctx context.Context, request reconcil
 	return reconcile.Result{}, nil
 }
 
+func getPrecedingVersion(instance *upgradev1alpha1.UpgradeConfig, clusterVersion *v1.ClusterVersion) string {
```

Comment:
Better to put function like this into `pkg/clusterversion/` ?

Reply:
done